### DISCUSSION TITLE
Can't fetch rows, returns empty array

### DISCUSSION THREAD

**Discussion by Clemens-B** [supabase/supabase#26798] [Questions]
Hello guys,

so I just created a table and tried to fetch some items from it like I do in many other tables. But I can not get any results, it is returning an empty array.

If I turn RLS off it works but I cannot get it working with RLS.

store/projects.js
```javascript
export async function getAllProjects() {
  const { data: projects, error } = await supabase
    .from("timeProjects")
    .select('*');
}
```

Policy
```bash
alter policy "Enable read access for all users"
on "public"."timeProjects"
to public
using (
  (auth.uid() = user_id)
);
```
DDL
![screenshot-supabase com-2024 05 27-15_35_26](https://github.com/supabase/supabase/assets/18402007/d74f4564-3d39-4cc1-b47e-c1f10b0bcd0d)

**Comment by GaryAustin1** [comment]
Change your RLS to TO anon and True for the policy as a test.
If that works then you don't have a user session in the client making the call.

**Comment by Clemens-B** [comment]
Thanks! Tested it, but no result.

```
alter policy "Enable read access for all users"
on "public"."timeProjects"
to anon
using (
  true
);
```
still empty array

**Reply by Clemens-B** [reply]
Deleted the policy for select statements and get also an empty array. Expected some RLS error

**Reply by GaryAustin1** [reply]
Are you using next.js?
It caches fetches in many cases.

**Reply by GaryAustin1** [reply]
Not having any RLS policy will not error.  It is silent on selects and will return empty array with no error.

**Reply by Clemens-B** [reply]
After re-adding the policy it somehow works. The code in my vue project etc. is the same as posted initially..

```shell
alter policy "Enable read access for all users"
on "public"."timeProjects"
to authenticated
using (
  (auth.uid() = user_id)
);
```

Do you validate/trim the user input for the policies?

Anyway, thanks a lot for your help!!

**Reply by GaryAustin1** [reply]
auth.uid() is from the jwt you have no control on that.  When you insert your data you should have RLS to make sure user_id = auth.uid() there too.
